release.config.js

```diff
@@ -0,0 +1,35 @@
+module.exports = {
+  branches: ['main'],
+  plugins: [
+    [
+      '@semantic-release/commit-analyzer',
+      {
+        preset: 'angular',
+        parserOpts: {
+          noteKeywords: ['BREAKING CHANGE', 'BREAKING CHANGES'],
+        },
+      },
+    ],
+    [
+      '@semantic-release/release-notes-generator',
+      {
+        preset: 'angular',
+      },
+    ],
+    [
+      '@semantic-release/npm',
+      {
+        npmPublish: true,
+      },
+    ],
+    ['@semantic-release/github'],
+    [
+      '@semantic-release/git',
+      {
+        assets: ['src/index.ts', 'docs', 'package.json'],
```

Comment:
Does src/index.ts need to be managed by the semantic-release bot?

Reply:
we need to make sure that we only ship the `lib` directory which is built after running `yarn prepare` or `bob build`.  and I dont think this current PR is accomplishing that.

Reply:
the `file` parameter in the Package.json determine which file we pushed to npm. Currently, I set "files": ["lib/**"], which means we only ship the files under the ilb directory.

Reply:
@kelsonpw 
Reference:https://github.com/semantic-release/npm/issues/128
And I saw the previously the files setting is as the following, which not including only the lib directory, which one should we use?

![](https://files.slack.com/files-pri/T02AKCA5Q-F01QY78HEDB/screen_shot_2021-03-11_at_10.15.12_pm.png)

Reply:
We want to follow that same list, you are right.  Thanks for catching that.  I think john is right about removing 'src/index.ts' from assets though